components/director/internal/domain/eventing/service_test.go

```diff
@@ -1040,25 +1017,31 @@ func Test_GetForApplication(t *testing.T) {
 		mock.AssertExpectationsForObjects(t, runtimeRepo, labelRepo)
 	})
 
-	t.Run("Error when ensuring the scenarios, there are no scenarios assigned to application", func(t *testing.T) {
+	t.Run("Empty configuration when there are no scenarios assigned to application", func(t *testing.T) {
 		// GIVEN
-		expectedError := fmt.Sprintf(`while ensuring the scenarios assigned to the runtime and application: while verifing whether runtime [ID=%s] belongs to the application scenarios: Internal Server Error: application does not belong to scenarios`, runtimeID)
 		ctx := fixCtxWithTenant()
+		emptyConfiguration := &model.ApplicationEventingConfiguration{
+			EventingConfiguration: model.EventingConfiguration{
+				DefaultURL: url.URL{},
+			},
+		}
 		runtimeRepo := &automock.RuntimeRepository{}
+		runtimePage := fixRuntimePageWithOne()
 		runtimeRepo.On("List", ctx, tenantID.String(), fixLabelFilterForRuntimeDefaultEventingForApp(),
-			1, mock.Anything).Return(fixRuntimePageWithOne(), nil)
+			1, mock.Anything).Return(runtimePage, nil)
 		labelRepo := &automock.LabelRepository{}
 		labelRepo.On("GetByKey", ctx, tenantID.String(), model.ApplicationLabelableObject,
 			applicationID.String(), model.ScenariosKey).Return(nil, apperrors.NewNotFoundError(resource.Label, ""))
+		labelRepo.On("Delete", ctx, tenantID.String(), model.RuntimeLabelableObject, runtimePage.Data[0].ID, fmt.Sprintf("%s_defaultEventing", applicationID.String())).Return(nil)
 
 		svc := NewService(nil, runtimeRepo, labelRepo)
 
 		// WHEN
-		_, err := svc.GetForApplication(ctx, app)
+		conf, err := svc.GetForApplication(ctx, app)
 
 		// THEN
-		require.Error(t, err)
-		require.EqualError(t, err, expectedError)
+		require.NoError(t, err)
+		require.Equal(t, emptyConfiguration, conf)
```

Comment:
This test relies on that the GetByKey returns NotFound and then on the second time on 187 line no runtime is found and therefore emptyConfig.

However, I think it makes sense to add a test that on the second execution on GetByKey returns a real runtime and therefore validate the full flow.

1. Get defatul eventing
2. See that there is no scenario anymore
3. Delete the default eventing
4. Get new runtime in scenario with the APP
5. Add default eventing to it
6. Return real config